gapii/cc/spy_base.h

```diff
@@ -237,6 +238,40 @@ inline Slice<T> SpyBase::slice(const Slice<T>& src, uint64_t s, uint64_t e) cons
     return src(s, e);
 }
 
+struct StateSerializer {
```

Comment:
This class confuses me on several levels.
What's `addr`? The address of a single state object, field?